extractor/filesystem/language/ruby/gemspec/gemspec.go

```diff
@@ -212,3 +249,662 @@ func extract(path string, r io.Reader) (*extractor.Package, error) {
 		PURLType: purl.TypeGem,
 	}, nil
 }
+
+func resolveVersionFromRequires(fsys fs.FS, gemspecPath string, requirePaths []string, constName string) (string, error) {
```

Comment:
I think it might be best to move this into its own file given how large it is

Reply:
Addressed with https://github.com/google/osv-scalibr/pull/1398/commits/daec86af437e661cd3ddcc3fd1c7482d12c3d1ec, happy to change the file name to something else, if needed.